### ISSUE TITLE
HELP！！Whether Linux can be installed in the Conda virtual environment？

### ISSUE THREAD

**Issue by eightfeetfish** [gpustack/gpustack#822]
The Python version in the base environment of my Linux server is 3.9, I need to install GPUStack, 
can I use conda create an environment with python version 3.10 to install it? Or do I need to upgrade the python version of the base environment and run "curl -sfL https://get.gpustack.ai|" in the base environment sh -“？

我Linux服务器中base环境下的Python版本是3.9，我需要部署GPUStack，我可以使用conda create一个python版本为3.10的环境进行安装吗？还是说我需要将base环境的python版本进行升级，在base环境中运行”curl -sfL https://get.gpustack.ai | sh -
“

**Comment by gitlawr** [comment]
You can use a conda env or other venv as long as the python version is supported(3.10-3.12).

**Comment by gitlawr** [comment]
And we recommend using the os distribution we've verified, see https://docs.gpustack.ai/latest/installation/installation-requirements/#operating-system-requirements

**Comment by eightfeetfish** [comment]
> You can use a conda env or other venv as long as the python version is supported(3.10-3.12).

So am I running the same command in the corresponding virtual environment?
”curl -sfL https://get.gpustack.ai/“

**Comment by gitlawr** [comment]
Yes, the same command. `curl -sfL https://get.gpustack.ai | sh -s -`

**Comment by eightfeetfish** [comment]
> Yes, the same command. `curl -sfL https://get.gpustack.ai | sh -s -`

Do I need to open which ports to use?such as 80 443?Are there any others~？

**Comment by gitlawr** [comment]
No.
The script installs with pip, then run GPUStack as a systemd service.

If you use `pip install gpustack[all]`, please refer to https://docs.gpustack.ai/0.4/installation/manual-installation/

**Comment by eightfeetfish** [comment]
> No. The script installs with pip, then run GPUStack as a systemd service.
> 
> If you use `pip install gpustack[all]`, please refer to https://docs.gpustack.ai/0.4/installation/manual-installation/

When I install using the command line, can I start the service via systemd?

**Comment by eightfeetfish** [comment]
> If I use ”pip install gpustack[all]" in the corresponding virtual environment, will it have the same effect as using the command line installation?



> > No. The script installs with pip, then run GPUStack as a systemd service.
> > If you use `pip install gpustack[all]`, please refer to https://docs.gpustack.ai/0.4/installation/manual-installation/
> 
> When I install using the command line, can I start the service via systemd?

I mean can you use systemd to start the UI?

**Comment by gitlawr** [comment]
Yes, an example is provided in the manual-installation link. The `gpustack start` command serves both API and UI.

**Comment by eightfeetfish** [comment]
> Yes, an example is provided in the manual-installation link. The `gpustack start` command serves both API and UI.
Can I specify the port on which the UI boots via gpustack start？

**Comment by eightfeetfish** [comment]
> Yes, an example is provided in the manual-installation link. The `gpustack start` command serves both API and UI.

Can I specify the port on which the UI boots via gpustack start

**Comment by eightfeetfish** [comment]
> Collaborator

Thank you very much!!!

**Comment by eightfeetfish** [comment]
> See https://docs.gpustack.ai/latest/installation/installation-requirements/#port-requirements

In the document: https://docs.gpustack.ai/latest/installation/installation-requirements/#port-requirements

Written:
The port range TCP 40000-41024 assigned to the inference service
The port range TCP 50000-51024 assigned to the llama-box RPC server

Is it enough to activate only one or more port numbers in the corresponding range? For example, I open port 40001 and 50001

**Comment by gitlawr** [comment]
No, ports are randomly assigned with the range. However, If you are getting started with a single node, you can skip them and only expose the management port( default 80 or specified by --port) until you use multiple nodes.

**Comment by eightfeetfish** [comment]
> No, ports are randomly assigned with the range. However, If you are getting started with a single node, you can skip them and only expose the management port( default 80 or specified by --port) until you use multiple nodes.

After I open all the ports I need, I don't need to manually deal with the port settings after the installation is successful, right?

**Comment by gitlawr** [comment]
Could you clarify what you mean by ‘manually deal with the port settings’?

**Comment by eightfeetfish** [comment]
> Could you clarify what you mean by ‘manually deal with the port settings’?

I mean, do you need to manually specify the port number and port number range in the UI interface or elsewhere?Or is it that the default port number will be automatically used when it is used after the installation is completed, and I don't need to go somewhere to configure it

**Comment by gitlawr** [comment]
You don't need to.

**Comment by eightfeetfish** [comment]
> No, you don't need to.

Thank you very much~！

**Comment by eightfeetfish** [comment]
I need to deploy GPUStack on the company's server, but the server cannot access the Internet and can only apply for the corresponding network segment permissions. If I use curl -sfL https://get.gpustack.ai | sh -s -
to install, do I only need to apply to connect to the network segment https://get.gpustack.ai?

**Comment by gitlawr** [comment]
When using the installation script you also need access to pypi.org or other pypi mirrors.
Please check https://docs.gpustack.ai/latest/installation/air-gapped-installation/ for air-gapped installation.
